app.py

```diff
@@ -184,28 +176,102 @@ def main():
     pull_cache_from_hub = arguments.pull_cache_from_hub
 
     # Initialize the interface and grab the UI-provided arguments
-    dataset_args = display_initial_UI()
-
-    # TODO: Make this less of a weird outlier.
-    show_perplexities = st.sidebar.checkbox("Show text perplexities")
-
-    # Initialize the main DMT class with the UI-provided arguments
-    # When using the app (this file), try to use cache by default.
-    dstats = dmt_cls(**dataset_args, use_cache=True)
-    display_title(dstats)
-    # Get the widget functionality for the different measurements.
-    load_prepare_list, display_list = get_widgets(dstats)
-    # Load/Prepare the DMT widgets.
-    loaded_dstats = load_or_prepare_widgets(dstats, load_prepare_list,
-                                            show_perplexities, live=live,
-                                            pull_cache_from_hub=pull_cache_from_hub)
-    # After the load_or_prepare functions are run,
-    # we should have a cache for each measurement widget --
-    # either because it was there already,
-    # or we computed them (which we do when not live).
-    # Write out on the UI what we have.
-    display_measurements(dataset_args, display_list, loaded_dstats,
-                         show_perplexities)
+    with gr.Blocks() as demo:
```

Comment:
Could all of the logic under this be its own function? It could be called `initialize_demo` or something like this.

Reply:
Yep good point - moved to a `create_demo` function!

Reply:
LGTM.